octane/evmengine/keeper/keeper.go

```diff
@@ -183,7 +188,10 @@ func (k *Keeper) parseAndVerifyProposedPayload(ctx context.Context, msg *types.M
 		return engine.ExecutableData{}, errors.Wrap(err, "eligible withdrawals")
 	}
 
-	if !withdrawalsEqual(payload.Withdrawals, eligibleWithdrawals) {
+	// If eligible withdrawals don't match with what's in the payload,
+	// only throw an error in strict mode or if the payload is non-empty
+	if !withdrawalsEqual(payload.Withdrawals, eligibleWithdrawals) &&
```

Comment:
I'd make this slightly more readable (also no need to pass in the flag, can infer from context):
```
// Allow 0 payload withdrawals for first block after drake upgrade.
	// Since block built and verified by magellen, but executed by drake.
	// So do strict validation unless in FinalizeBlock and payload is empty.
	strictWithdrawals := sdk.UnwrapSDKContext(ctx).ExecMode() == sdk.ExecModeProcessProposal || len(payload.Withdrawals) > 0

	if !withdrawalsEqual(payload.Withdrawals, eligibleWithdrawals) && strictWithdrawals {
		return engine.ExecutableData{}, errors.New("mismatch with eligible withdrawals")
	}
```